### DISCUSSION TITLE
Before and after hooks function signature

### DISCUSSION THREAD

**Discussion by servohatred** [artilleryio/artillery#2554] [Artillery]
There is information in the docs about before/afterScenario and before/afterRequest hooks method signatures: https://www.artillery.io/docs/reference/engines/http#loading-custom-code
But there is no info about method signatures for top level hooks before and after:
https://www.artillery.io/docs/reference/test-script#all-engines-supported
I am trying to create a before hook  using playwright engine but I am not sure what parameters should I use. Should I have a next() ?
Can you help me please?  :) thank you!

**Comment by bernardobridge** [comment]
Hi @servohatred 👋 

The `before` and `after` section just run a scenario of the specified engine, once before all others. So if you want a Playwright `before` to run, you would use the same signature of writing a Playwright scenario (check [docs](https://www.artillery.io/docs/reference/engines/playwright#test-function-api) for PW function signature). If you don't specify the `engine`, it will default to `http` (same as scenarios), which would allow you to use an HTTP engine `before` hook.

There's an example of using a Playwright `before` here: https://github.com/artilleryio/artillery/tree/main/examples/browser-playwright-reuse-authentication

Does that make sense?

**Reply by servohatred** [reply]
Thank you so much for the explanation!
